packages/samples/headless-react/src/components/result-list/result-list.fn.tsx

```diff
@@ -0,0 +1,55 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {
+  buildResultList,
+  ResultList as HeadlessResultList,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface ResultListProps {
+  controller: HeadlessResultList;
+}
+
+export const ResultList: FunctionComponent<ResultListProps> = (props) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  const showMore = () => {
+    controller.fetchMoreResults();
+  };
+
+  const resultsAvailable = () => {
+    return state.results.length > 0;
+  };
+
+  const results = () => {
+    return state.results.map((result) => (
+      <li key={result.uniqueId}>
+        <article>
+          <h2>
+            <a href={result.clickUri}>{result.title}</a>
+          </h2>
+          <p>{result.excerpt}</p>
+        </article>
+      </li>
+    ));
+  };
+
+  return (
+    <div>
+      <ul>{resultsAvailable() ? results() : <li>No results</li>}</ul>
+      {resultsAvailable() && (
+        <button onClick={() => showMore()} disabled={state.isLoading}>
+          Show more
+        </button>
```

Comment:
Same comment as above on removing the `Show more`, and grouping all the rendered elements in one place so that readers do not need to scroll up and down.